### PR TITLE
feat: add multi-document YAML support and update documentation

### PR DESCRIPTION
**Pull request by Romain7495**
fixes: https://github.com/ddebin/yaml-sort/issues/23

- Add support for multiple YAML documents in a single file
- Add support for invalid yaml
- Add test cases for multi-document YAML handling
- Add test cases for invalid yaml (duplicared keys)
- Add test for quoted values (force)
- Update README.md with forceQuotes option documentation

### CODE REVIEW

File: yaml-sort.js [hunk 1]
```diff
@@ -88,28 +93,56 @@ argv.input.forEach((file) => {
...
+      const originalContent = (hasDocumentStart ? '---\n' : '') + documents
+        .map(doc => yaml.dump(doc, {
+          indent: argv.indent,
+          lineWidth: argv.lineWidth,
+          quotingType: argv.quotingStyle === 'double' ? '"' : "'",
+          forceQuotes: argv.forceQuotes
+        }))
+        .join('---\n')
+
+      if (sortedContent !== originalContent) {
```
Comment (ddebin) [inline yaml-sort.js:138]: @Romain7495 Why doesn't it check `sortedContent` with `content.toString()` like before?

File: yaml-sort.js [hunk 2]
```diff
@@ -88,28 +93,56 @@ argv.input.forEach((file) => {
...
-        argv.stdout || (argv.output === '.') || (isStdin && !argv.output)
-          ? process.stdout.fd
-          : (argv.output ? argv.output : file)
+      argv.stdout || (argv.output === '.') || (isStdin && !argv.output)
+        ? process.stdout.fd
+        : (argv.output ? argv.output : file)
 
     const content = fs.readFileSync(isStdin ? process.stdin.fd : file, argv.encoding)
+    let documents
+    try {
```
Comment (ddebin) [inline yaml-sort.js:102]: @Romain7495 This code is already within a try/catch.

File: yaml-sort.js (hunk 1 above)
Comment (Romain7495) [reply yaml-sort.js:138]: Hello @ddebin, there is an edge case, when you have a key in you yaml `y: foo`, `js-yaml` lib will wrap it into `'y'` because `y` is ambiguous. If we compare with `toString`, it will compare `'y': foo` with `y: foo`

I have added two test cases that verify this in `test/test-multiple-sorted.yml` and `test/test-multiple.yml` files

File: yaml-sort.js (hunk 2 above)
Comment (Romain7495) [reply yaml-sort.js]: totally agree sorry, I removed it, but I have added a test case for duplicated keys

**Comment by Romain7495** [comment]
hello @ddebin any news for this PR ?

**Comment by ddebin** [comment]
> hello @ddebin any news for this PR ?

I will have a look tomorrow.

File: yaml-sort.js [hunk 3]
```diff
@@ -88,28 +93,48 @@ argv.input.forEach((file) => {
...
+      argv.stdout || (argv.output === '.') || (isStdin && !argv.output)
+        ? process.stdout.fd
+        : (argv.output ? argv.output : file)
 
     const content = fs.readFileSync(isStdin ? process.stdin.fd : file, argv.encoding)
 
-    const sorted = yaml.dump(yaml.load(content), {
+    const documents = yaml.loadAll(content)
+
+    if (!documents) {
```
Comment (ddebin) [inline yaml-sort.js:104]: @Romain7495 Can't trigger this with a test. I think it's useless as it can't be triggered.

File: yaml-sort.js (hunk 1 above)
Comment (ddebin) [reply yaml-sort.js:130]: @Romain7495 
Replaced `if (sortedContent !== originalContent)` with `if (sortedContent !== content.toString())` and it still passes tests...

Comment (Romain7495) [reply yaml-sort.js]: well ok, it's fixed

File: yaml-sort.js (hunk 3 above)
Comment (Romain7495) [reply yaml-sort.js]: I removed the condition, It was relevant only why my try catch, but removed as well

**Comment by ddebin** [comment]
Thanks for the PR! Merged!
